### ISSUE TITLE
Automatically set tls_maximum_protocol_version to opt in to TLS 1.3

### ISSUE THREAD

**Issue by cromefire** [heptio/contour#672]
Please add support for the `tls_maximum_protocol_version` envoy api parameter (like you do for `tls_minimum_protocol_version`) to allow users to opt in to TLS 1.3. See https://github.com/envoyproxy/envoy/issues/3544.

**Comment by cromefire** [comment]
How should that condition look like? Just annotation to enable TLS 1.3? I might try doing that myself, once envoy is updated.

**Comment by cromefire** [comment]
Okay just tell me when you are done with 1.7.x and I will start try around with go

**Comment by davecheney** [comment]
Blocked on #650 / #443

**Comment by davecheney** [comment]
@cromefire the blocking issues have been resolved. You're welcome to work on this for 0.7 which will go into free in the last week of October.

**Comment by cromefire** [comment]
Allright, it would be nice if you could assign, so so I don't loose track of it

**Comment by davecheney** [comment]
I tried but github won't let me.

**Comment by cromefire** [comment]
I gave up due to problems with the sign off constraint and go (and godep) being bad at being user friendly.

So if some one else want's to look at it, the solutions are the but I'm unable to commit them without trying again and again.

**Comment by rosskukulinski** [comment]
@cromefire sorry about the issues with the sign-off checker.  It should be working now if you wouldn't mind opening up one more new PR.
